### PR TITLE
Feature/continious integration

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,174 @@
...
+ 2. SDL build without unit tests on 3 policy flows (triggers: push, nightly)
+ 3. Smoke tests on 3 policy flows (triggers: build jobs without unit tests)
+ 4. Feature tests on 3 policy flows (triggers: build jobs without unit tests)
+ 5. Regression tests on 3 policy flows (triggers: build jobs without unit tests)
+
+## Special requests
+
+In case of special requiest (not ordinaary builds for special feature, restart certain jobs, etc, ...) please contact any of following persons:
+ - [Yarik Mamykin](https://github.com/YarikMamykin) : akutsan@luxoft.com
+ - [Alexander Kutsan](https://github.com/LuxoftAKutsan) : YMamykin@luxoft.com
```
Comment (YarikMamykin) [inline README.md]: Please swap emails

File: README.md [hunk 2]
```diff
@@ -1,2 +1,174 @@
...
+After job execution will be created a view with the following checks:
+ 1. SDL build with unit tests on 3 policy flows (triggers: push, nightly)
+ 2. SDL build without unit tests on 3 policy flows (triggers: push, nightly)
+ 3. Smoke tests on 3 policy flows (triggers: build jobs without unit tests)
+ 4. Feature tests on 3 policy flows (triggers: build jobs without unit tests)
+ 5. Regression tests on 3 policy flows (triggers: build jobs without unit tests)
+
+## Special requests
+
+In case of special requiest (not ordinaary builds for special feature, restart certain jobs, etc, ...) please contact any of following persons:
```
Comment (YarikMamykin) [inline README.md]: Please fix typos `requiest` and `ordinaary`

**Review by YarikMamykin — CHANGES_REQUESTED** [review]
Generally excellent work!

File: README.md (hunk 2 above)
Comment (YarikMamykin) [reply README.md]: @YarikMamykin done in https://github.com/LuxoftSDL/sdl_core_ci/pull/1/commits/a280e8231a579ed768b23a5aa05ba0f1c1e3b2e6

File: README.md (hunk 1 above)
Comment (YarikMamykin) [reply README.md]: @YarikMamykin done in https://github.com/LuxoftSDL/sdl_core_ci/pull/1/commits/a280e8231a579ed768b23a5aa05ba0f1c1e3b2e6

**Review by YarikMamykin — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -1,2 +1,174 @@
...
+
+CI strategy described in details in [proposal](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0277-Continuous-Integration-And-Testing.md)
+
+## Develop nightly and push checks
+
+Only TCP transport is checked on 3 policy flows. 
+
+[Develop](https://github.com/smartdevicelink/sdl_core/tree/develop) push and nightly builds are available in [view](https://opensdl-jenkins.prjdmz.luxoft.com/view/Develop_push_and_nightly_status/) :
+
+ - [Code style check](https://opensdl-jenkins.prjdmz.luxoft.com/view/all/job/Develop_SDL_Checkstyle/) : Use [check_style.sh](https://github.com/smartdevicelink/sdl_core/blob/master/tools/infrastructure/check_style.sh) to check sdl_core code for compilence to Google coding style. ![bage][check style bage]
```
Comment (dboltovskyi) [inline README.md]: Typo `compilence`

File: README.md [hunk 4]
```diff
@@ -1,2 +1,174 @@
...
+https://opensdl-jenkins.prjdmz.luxoft.com/view/all/job/Develop_SDL_NoUT_E_BWSS_OFF/ 
+
+Full weekly status is available on https://opensdl-jenkins.prjdmz.luxoft.com/view/Develop_weekly_status/ 
+
+## Feature checks:
+
+For each feature before merging to develop should be created a list of jobs similar to develop to check that feature will no introduce a regression. 
+There is a special job in CI [Feature job create]() that will create a list of jobs and a separate view for the feature.
+
+Required input values for feeature job:
```
Comment (dboltovskyi) [inline README.md]: Typo `feeature`

File: README.md [hunk 5]
```diff
@@ -1,2 +1,174 @@
...
+
+Full weekly status is available on https://opensdl-jenkins.prjdmz.luxoft.com/view/Develop_weekly_status/ 
+
+## Feature checks:
+
+For each feature before merging to develop should be created a list of jobs similar to develop to check that feature will no introduce a regression. 
+There is a special job in CI [Feature job create]() that will create a list of jobs and a separate view for the feature.
+
+Required input values for feeature job: 
+ - Feture name (will be used for view title)
```
Comment (dboltovskyi) [inline README.md]: Typo `Feture`

File: README.md [hunk 6]
```diff
@@ -1,2 +1,174 @@
...
+ - sdl_core feature branch and repository (master by default) 
+ - sdl_atf feature branch and repository (master by default)
+ - sdl_atf_test_scripts deature branch and repository. (master by default)
```
Comment (dboltovskyi) [inline README.md]: Typo `deature`

File: README.md [hunk 7]
```diff
@@ -1,2 +1,174 @@
...
+ - Feature test set (optional)
+ - Additional info : evoluiton proposal, links to issues, etc ...
```
Comment (dboltovskyi) [inline README.md]: Typo `evoluiton`

File: README.md [hunk 8]
```diff
@@ -1,2 +1,174 @@
-# sdl_core_ci
-Repository for housing continuous integration related scripts and files for SDL Core
+# SDL continuous integration strategy
+
+SDL continuously checks code:
+ - check style
+ - static code analysis
+ - Build and unit tests
+ - Smoke automates tests
+ - Regression tests
```
Comment (dboltovskyi) [inline README.md]: All bullet points should start with upper (or lower) case

File: README.md [hunk 9]
```diff
@@ -1,2 +1,174 @@
-# sdl_core_ci
-Repository for housing continuous integration related scripts and files for SDL Core
+# SDL continuous integration strategy
+
+SDL continuously checks code:
+ - check style
+ - static code analysis
+ - Build and unit tests
+ - Smoke automates tests
```
Comment (dboltovskyi) [inline README.md]: Typo `automates`

File: README.md (hunk 8 above)
Comment (dboltovskyi) [inline README.md]: Regression automated tests?

File: README.md [hunk 10]
```diff
@@ -1,2 +1,174 @@
...
+ - Regression tests
+
+Job artifacts contains build results, ATF reports, SDL logs,  etc ...
```
Comment (dboltovskyi) [inline README.md]: Typo `contain`

File: README.md [hunk 11]
```diff
@@ -1,2 +1,174 @@
...
+
+ - [Code style check](https://opensdl-jenkins.prjdmz.luxoft.com/view/all/job/Develop_SDL_Checkstyle/) : Use [check_style.sh](https://github.com/smartdevicelink/sdl_core/blob/master/tools/infrastructure/check_style.sh) to check sdl_core code for compilence to Google coding style. ![bage][check style bage] 
+ - Develop builds without unit tests in 3 policy flows: 
+   - [![bage][develop proprietary no ut bage]][develop proprietary no ut] 
+   - [![bage][develop external proprietary no ut bage]][develop external proprietary no ut] 
+   - [![bage][develop http no ut bage]][develop http no ut] 
+ - Develop builds and unit unit tests run: 
+   - [ ![bage][develop proprietary ut bage]][develop proprietary ut]
+   - [![bage][develop external proprietary ut bage]][develop external proprietary ut] 
+   - [![bage][develop http ut bage]][develop http ut]
```
Comment (dboltovskyi) [inline README.md]: All policy flows I would name as:
 - `PROPRIETARY`
 - `EXTERNAL_PROPRIETARY`
 - `HTTP`

Everywhere in doc

File: README.md [hunk 12]
```diff
@@ -1,2 +1,174 @@
...
+   - [![bage][develop http no ut bage]][develop http no ut] 
+ - Develop builds and unit unit tests run: 
+   - [ ![bage][develop proprietary ut bage]][develop proprietary ut]
+   - [![bage][develop external proprietary ut bage]][develop external proprietary ut] 
+   - [![bage][develop http ut bage]][develop http ut] 
+ - [Develop_=RUN_PUSH_AND_NIGHTLY=](
+https://opensdl-jenkins.prjdmz.luxoft.com/view/Develop%20push%20and%20nightly/job/Develop_=RUN_PUSH_AND_NIGHTLY=/) job is trigger for listed develop build jobs
+
+
+Develop branch push and nightly automated scripts checks available in [view](https://opensdl-jenkins.prjdmz.luxoft.com/view/all/) :
```
Comment (dboltovskyi) [inline README.md]: Provided link is for `all` jobs, not for `develop`

File: README.md [hunk 13]
```diff
@@ -1,2 +1,174 @@
...
+List of triggered jobs available in each build job as Downstream projects, example:
+https://opensdl-jenkins.prjdmz.luxoft.com/view/all/job/Develop_SDL_NoUT_E_BWSS_OFF/ 
+
+Full weekly status is available on https://opensdl-jenkins.prjdmz.luxoft.com/view/Develop_weekly_status/ 
+
+## Feature checks:
+
+For each feature before merging to develop should be created a list of jobs similar to develop to check that feature will no introduce a regression. 
+There is a special job in CI [Feature job create]() that will create a list of jobs and a separate view for the feature.
+
```
Comment (dboltovskyi) [inline README.md:144]: Let's put a link to https://opensdl-jenkins.prjdmz.luxoft.com/view/Jenkins_Utils/ and some description for all three jobs

File: README.md [hunk 14]
```diff
@@ -1,2 +1,174 @@
...
+ 4. Feature tests on 3 policy flows (triggers: build jobs without unit tests)
+ 5. Regression tests on 3 policy flows (triggers: build jobs without unit tests)
+
+## Special requests
+
+In case of special request (not ordinary builds for special feature, restart certain jobs, etc, ...) please contact any of following persons:
+ - [Yarik Mamykin](https://github.com/YarikMamykin) : YMamykin@luxoft.com
+ - [Alexander Kutsan](https://github.com/LuxoftAKutsan) : akutsan@luxoft.com
+
+[check style bage]: https://img.shields.io/jenkins/build?jobUrl=https%3A%2F%2Fopensdl-jenkins.prjdmz.luxoft.com%2Fview%2Fall%2Fjob%2FDevelop_SDL_Checkstyle%2F&label=check%20style
```
Comment (dboltovskyi) [inline README.md:166]: Is this correct link?

File: README.md [hunk 15]
```diff
@@ -1,2 +1,174 @@
...
+[proprietary atf policy bage]:https://img.shields.io/jenkins/tests?jobUrl=https%3A%2F%2Fopensdl-jenkins.prjdmz.luxoft.com%2Fview%2Fall%2Fjob%2FDevelop_TCP_ATF_Policies_P%2F&label=proprietary%20policy%20tests
+
+[external proprietary atf policy]: https://opensdl-jenkins.prjdmz.luxoft.com/view/all/job/Develop_TCP_ATF_Policies_E/
+[external proprietary atf policy bage]:https://img.shields.io/jenkins/tests?jobUrl=https%3A%2F%2Fopensdl-jenkins.prjdmz.luxoft.com%2Fview%2Fall%2Fjob%2FDevelop_TCP_ATF_Policies_E%2F&label=external%20proprietary%20policy%20tests
+
+
+[http atf policy]: https://opensdl-jenkins.prjdmz.luxoft.com/view/all/job/Develop_TCP_ATF_Policies_H/
+[http atf policy bage]:https://img.shields.io/jenkins/tests?jobUrl=https%3A%2F%2Fopensdl-jenkins.prjdmz.luxoft.com%2Fview%2Fall%2Fjob%2FDevelop_TCP_ATF_Policies_H%2F&label=http%20policy%20tests
+
+[unit test coverage]: https://opensdl-jenkins.prjdmz.luxoft.com/view/all/job/develop_weekly_coverage/
```
Comment (dboltovskyi) [inline README.md]: Link is broken

**Review by dboltovskyi — CHANGES_REQUESTED** [review]

File: README.md [hunk 16]
```diff
@@ -97,16 +97,18 @@ Full weekly status is available on https://opensdl-jenkins.prjdmz.luxoft.com/vie
 
 ## Feature checks:
 
-For each feature before merging to develop should be created a list of jobs similar to develop to check that feature will no introduce a regression. 
-There is a special job in CI [Feature job create]() that will create a list of jobs and a separate view for the feature.
+For each feature before merging to develop should be created a list of jobs similar to develop to check that feature will not introduce a regression. 
+There is a special job on CI [Feature job create](https://opensdl-jenkins.prjdmz.luxoft.com/view/Jenkins_Utils/job/Jenkins_Utils_Create_Jobs_for_Feature/) that will create a list of jobs and a separate view for the feature.
+When feature is delivered and no checks for it needed appropriate jobs on CI can be deleted with [Feature job delete](https://opensdl-jenkins.prjdmz.luxoft.com/view/Jenkins_Utils/job/Jenkins_Utils_Delete_Jobs_for_Feature/)
```
Comment (dboltovskyi) [inline README.md]: `... no checks for it is needed ...`

File: README.md [hunk 17]
```diff
@@ -123,52 +125,52 @@ In case of special request (not ordinary builds for special feature, restart cer
...
+[develop PROPRIETARY no ut]: https://opensdl-jenkins.prjdmz.luxoft.com/view/all/job/Develop_SDL_NoUT_P/
+[develop PROPRIETARY no ut bage]:
+https://img.shields.io/jenkins/build?jobUrl=https%3A%2F%2Fopensdl-jenkins.prjdmz.luxoft.com%2Fview%2Fall%2Fjob%2FDevelop_SDL_NoUT_P%2F&label=PROPRIETARY%20build%20%20no%20UT
 
-[develop external proprietary no ut]: https://opensdl-jenkins.prjdmz.luxoft.com/view/all/job/Develop_SDL_NoUT_E/
-[develop external proprietary no ut bage]:
-https://img.shields.io/jenkins/build?jobUrl=https%3A%2F%2Fopensdl-jenkins.prjdmz.luxoft.com%2Fview%2Fall%2Fjob%2FDevelop_SDL_NoUT_E%2F&label=external%20proprietary%20build%20%20no%20UT
+[develop EXTERNAL PROPRIETARY no ut]: https://opensdl-jenkins.prjdmz.luxoft.com/view/all/job/Develop_SDL_NoUT_E/
+[develop EXTERNAL PROPRIETARY no ut bage]:
+https://img.shields.io/jenkins/build?jobUrl=https%3A%2F%2Fopensdl-jenkins.prjdmz.luxoft.com%2Fview%2Fall%2Fjob%2FDevelop_SDL_NoUT_E%2F&label=external%20PROPRIETARY%20build%20%20no%20UT
```
Comment (dboltovskyi) [inline README.md]: `label=external`,  here and everywhere needs to be corrected
also `label=http%20unit` and others

File: README.md [hunk 18]
```diff
@@ -55,9 +55,9 @@ A test set executed for SDL build in 3 policy flows:
...
 
 A test set executed for SDL build in 3 policy flows: 
- - [![bage][proprietary atf policy bage]][proprietary atf policy ]
- - [![bage][external proprietary atf policy bage]][external proprietary atf policy ]
- - [![bage][http atf policy bage]][http atf policy]
+ - [![bage][PROPRIETARY atf policy bage]][PROPRIETARY atf policy ]
+ - [![bage][EXTERNAL PROPRIETARY atf policy bage]][EXTERNAL PROPRIETARY atf policy ]
+ - [![bage][HTTP atf policy bage]][HTTP atf policy]
 
 #### Automated RC regression
```
Comment (dboltovskyi) [inline README.md:95]: Do we need to add some links here?

File: README.md [hunk 19]
```diff
@@ -1,2 +1,176 @@
...
+[PROPRIETARY atf policy bage]:https://img.shields.io/jenkins/tests?jobUrl=https%3A%2F%2Fopensdl-jenkins.prjdmz.luxoft.com%2Fview%2Fall%2Fjob%2FDevelop_TCP_ATF_Policies_P%2F&label=PROPRIETARY%20policy%20tests
+
+[EXTERNAL PROPRIETARY atf policy]: https://opensdl-jenkins.prjdmz.luxoft.com/view/all/job/Develop_TCP_ATF_Policies_E/
+[EXTERNAL PROPRIETARY atf policy bage]:https://img.shields.io/jenkins/tests?jobUrl=https%3A%2F%2Fopensdl-jenkins.prjdmz.luxoft.com%2Fview%2Fall%2Fjob%2FDevelop_TCP_ATF_Policies_E%2F&label=external%20PROPRIETARY%20policy%20tests
+
+
+[HTTP atf policy]: https://opensdl-jenkins.prjdmz.luxoft.com/view/all/job/Develop_TCP_ATF_Policies_H/
+[HTTP atf policy bage]:https://img.shields.io/jenkins/tests?jobUrl=https%3A%2F%2Fopensdl-jenkins.prjdmz.luxoft.com%2Fview%2Fall%2Fjob%2FDevelop_TCP_ATF_Policies_H%2F&label=http%20policy%20tests
+
+[unit test coverage]: https://opensdl-jenkins.prjdmz.luxoft.com/view/Develop_weekly/job/Develop_SDL_Checkstyle/
```
Comment (dboltovskyi) [inline README.md]: Is this correct job for UT coverage?

**Review by dboltovskyi — CHANGES_REQUESTED** [review]

**Review by dboltovskyi — APPROVED** [review]

**Pull request merged**